presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeJoinQueries.java

```diff
@@ -120,6 +120,22 @@ public void testMergeJoin()
         assertQuery(mergeJoin(), sql, getSession(), sql);
     }
 
+    @Test
+    public void testJoinsWithConstantFilterAndNoCriteria()
+    {
+        String query = "SELECT count(*) FROM (SELECT * FROM lineitem WHERE orderkey %% 1024 = 0) "
+                + "lineitem %s JOIN (SELECT * FROM orders WHERE orderkey %% 1024 = 0) "
+                + "orders ON orders.orderkey = 1024";
```

Comment:
Given this PR is to add support for left/right joins **without criteria**, this query still has a join criteria(or join condition: `ON orders.orderkey = 1024`) right?

Reply:
Here we are trying to test when join criteria is empty but there is a constant filter (node->filter).